cmd/cluster-agent/api/v1/language_detection.go

```diff
@@ -25,6 +26,12 @@ func InstallLanguageDetectionEndpoints(r *mux.Router) {
 }
 
 func postDetectedLanguages(w http.ResponseWriter, r *http.Request) {
+	if !config.Datadog.GetBool("language_detecion.enabled") {
+		languagedetection.ErrorResponses.Inc()
+		http.Error(w, "Language detection feature is disabled on the cluster agent", http.StatusInternalServerError)
```

Comment:
HTTP has the return code 503 for when the service is unavailable. I wonder if it would be useful here.